repository_service_tuf_worker/repository.py

```diff
@@ -368,15 +375,18 @@ def _update_snapshot(
 
         return snapshot.signed.version
 
-    def _get_path_succinct_role(self, target_path: str) -> str:
+    def _get_role_for_target_path(self, target_path: str) -> str:
         """
         Return role name by target file path
         """
-        bin_role: Metadata[Targets] = self._storage_backend.get(Targets.type)
-        bin_succinct_roles = bin_role.signed.delegations.succinct_roles
-        bins_name = bin_succinct_roles.get_role_for_target(target_path)
-
-        return bins_name
+        targets: Metadata[Targets] = self._storage_backend.get(Targets.type)
+        delegations = targets.signed.delegations
+        # Note: there could be more than one matching custom delegated role,
+        # for this target path, but we get the first match. If we want to get
+        # the most specific one we need to use _preorder_depth_first_walk
+        # of the ngclient Updater class in python-tuf.
+        role_name, _ = next(delegations.get_roles_for_target(target_path))
```

Comment:
See my comment below
```suggestion
        try:
            role_name, _ = next(delegations.get_roles_for_target(target_path))
        except StopIteration:
            return None
```